### PR TITLE
Add tests tolerance and comparator

### LINKED ISSUE #40: Vulkan: Implement DoTolerance command
Implement the DoTolerance command in the Vulkan engine.

### PR DESCRIPTION
**Pull request by jaebaek**
Fixes #40

### CODE REVIEW

File: src/verifier.cc [hunk 1]
```diff
@@ -28,18 +28,21 @@ const double kDefaultTexelTolerance = 0.002;
...
                           double tolerance,
                           const bool is_tolerance_percent = true) {
   double difference = std::fabs(real - expected);
-  if (difference == 0.0)
-    return true;
-  if (is_tolerance_percent)
-    tolerance *= difference;
-  return difference < tolerance;
+  if (is_tolerance_percent) {
+    if (difference > tolerance / 100.0 * std::fabs(expected))
```
Comment (dj2) [inline src/verifier.cc]: Can you add some bracketing here? Makes it easier to see it's (tolerance / 100.0) * std::fabs(expected)

File: src/verifier.cc [hunk 2]
```diff
@@ -28,18 +28,21 @@ const double kDefaultTexelTolerance = 0.002;
...
+      return false;
+  } else {
+    if (difference > tolerance)
```
Comment (dj2) [inline src/verifier.cc]: make it } else if (diff ...) {

File: src/verifier.cc (hunk 1 above)
Comment (jaebaek) [reply src/verifier.cc]: done

File: src/verifier.cc (hunk 2 above)
Comment (jaebaek) [reply src/verifier.cc]: done

**Pull request merged**